### ISSUE TITLE
[Accessibility]: User settings menu takes up valuable screen estate all the time

### ISSUE THREAD

**Issue by kopancek** [sourcegraph/sourcegraph#35706]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34114

### Problem description

If you navigate to user settings on a mobile device, it is always shown. Meaning that instead of showing the content of the page - we are taking more than half of the screen by the menu and the user needs to scroll horizontally to reach the content. 
![Uploading Screenshot 2022-05-19 at 14.49.52.png…]()

### Expected behavior

Show the menu as a dropdown with a hamburger trigger button, similar to how all the other websites in the world handle large menus on mobile devices.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by sourcegraph-bot-2** [comment]
Heads up @taylorsperry @jasongornall - the "team/frontend-platform" label was applied to this issue.

**Comment by umpox** [comment]
Related to https://github.com/sourcegraph/sourcegraph/issues/35706

**Comment by limitedmage** [comment]
We need to quickly address this for WCAG by moving the sidebar to the top. We can work later on refining the UI of the page to use a better design (eg. hamburger menu).
